### PR TITLE
Update rancher pull-scripts rule

### PR DESCRIPTION
**Pull request by egibs**
This PR addresses the continued critical finding as seen in: https://github.com/wolfi-dev/os/pull/31009

The current test data was generated when this rule still applied so there weren't any refresh updates.

### CODE REVIEW

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: rules/false_positives/rancher.yara [hunk 1]
```diff
@@ -3,11 +3,12 @@ rule pull_scripts : override {
     curl_chmod_relative_run_tiny = "medium"
     description = "pull-scripts"
   strings:
-    $binary = "BINARY_NAME=\"charts-build-scripts_${OS}_${ARCH}.exe\""
-    $chmod = "chmod +x ./bin/charts-build-scripts"
-    $echo1 = "echo \"Downloading charts-build-scripts version ${CHARTS_BUILD_SCRIPTS_REPO}@${CHARTS_BUILD_SCRIPT_VERSION}\""
-    $echo2 = "echo \"${BINARY_NAME} => ./bin/charts-build-scripts\""
-    $version_cmd = "./bin/charts-build-scripts --version"
+    $downloading = "echo \"Downloading charts-build-scripts version ${CHARTS_BUILD_SCRIPTS_REPO}@${CHARTS_BUILD_SCRIPT_VERSION}\""
```
[the remark this answers is not in the document]
Comment (egibs) [reply rules/false_positives/rancher.yara]: Updated in [`7d2c6b1` (#528)](https://github.com/chainguard-dev/malcontent/pull/528/commits/7d2c6b1556637a8cdd489855abb0de4eddcbf52d).

File: rules/false_positives/rancher.yara [hunk 2]
```diff
@@ -3,11 +3,11 @@ rule pull_scripts : override {
...
   strings:
-    $binary = "BINARY_NAME=\"charts-build-scripts_${OS}_${ARCH}.exe\""
+    $binary_name = "BINARY_NAME=charts-build-scripts"
+    $charts = "charts-build-scripts"
     $chmod = "chmod +x ./bin/charts-build-scripts"
-    $echo1 = "echo \"Downloading charts-build-scripts version ${CHARTS_BUILD_SCRIPTS_REPO}@${CHARTS_BUILD_SCRIPT_VERSION}\""
-    $echo2 = "echo \"${BINARY_NAME} => ./bin/charts-build-scripts\""
-    $version_cmd = "./bin/charts-build-scripts --version"
+    $echo_name = "echo \"${BINARY_NAME} => ./bin/charts-build-scripts\""
+    $version = "./bin/charts-build-scripts --version"
```
Comment (tstromberg) [inline rules/false_positives/rancher.yara:10]: these last two rules duplicate $charts

File: rules/false_positives/rancher.yara [hunk 3]
```diff
@@ -3,11 +3,11 @@ rule pull_scripts : override {
     curl_chmod_relative_run_tiny = "medium"
     description = "pull-scripts"
   strings:
-    $binary = "BINARY_NAME=\"charts-build-scripts_${OS}_${ARCH}.exe\""
+    $binary_name = "BINARY_NAME=charts-build-scripts"
+    $charts = "charts-build-scripts"
```
Comment (tstromberg) [inline rules/false_positives/rancher.yara:7]: consider adding `fullword` declaration so that this doesn't match zcharts-build-scripts

File: rules/false_positives/rancher.yara [hunk 4]
```diff
@@ -3,11 +3,11 @@ rule pull_scripts : override {
     curl_chmod_relative_run_tiny = "medium"
     description = "pull-scripts"
   strings:
-    $binary = "BINARY_NAME=\"charts-build-scripts_${OS}_${ARCH}.exe\""
+    $binary_name = "BINARY_NAME=charts-build-scripts"
```
Comment (tstromberg) [inline rules/false_positives/rancher.yara:6]: I think this $binary_name rule should be sufficient - it's a superset of $charts.

**Review by tstromberg — CHANGES_REQUESTED** [review]

**Comment by tstromberg** [comment]
TL;DR: Let's start with simpler (yet still unlikely) overrides when possible.

**Review by tstromberg — APPROVED** [review]
